accepted/2021/install-location-per-architecture.md

```diff
@@ -204,17 +230,28 @@ for .NET 3.1 (which will read the non-architecture-specific variable only).
 
 ## Alternative designs
 
-### Configuration file per architecture
+### One configuration file with multiple lines
+
+In this case the `/etc/dotnet/install_location` file format is extended to support
+multiple lines with architecture prefixes, like:
+
+```console
+/usr/local/share/dotnet/x64
+arm64=/usr/local/share/dotnet
+x64=/usr/local/share/dotnet/x64
+```
+
+The advantage is less files in the configuration folder and arguably easier to
+understand configuration for humans.
+
+The main downside is added complexity to the product and installers.
 
-In this case x64 apphost would look for its configuration in `/etc/dotnet/install_location_x64`.
-And `arm64` apphost would look into `/etc/dotnet/install_location_arm64`.
-Both would fallback to the existing `/etc/dotnet/install_location` if the arch-specific
-one doesn't exist.
+Apphosts would have to contain a parser for this format and implement additional
+logic to handle duplicate entries and such.
 
-The effect of this is almost identical to the proposed solution.
-The only downside is that this doesn't follow to "native first" approach
-where the native architecture gets the existing "nice" names. It also adds more
-files into `/etc/dotnet` which doesn't feel necessary.
+But more importantly installers would have to be able to read and edit this file
+which is rather challenging given the currently used installer technologies
```

Comment:
I wouldn't call this a limitation of the installer technology.  All installer technologies have primitives they support (files, registry keys, environment, etc).  When you design a product with installation state that doesn't easily map to one of those primitives it requires a custom action.  Any custom action is difficult, the complexity of the custom action makes it more difficult.  Adding ambiguous state (eg: mix of user/product) makes it more difficult.  Making it a shared component between multiple products makes it more difficult, especially when those products have a desired state that is different than just exists.

The general goal for installers it to get them as simple as possible.  Make it so that they are unlikely to fail and deal in very simple primitives.